WMF Framework/MediaWikiFetcher.swift

```diff
@@ -83,9 +83,51 @@ public final class MediaWikiFetcher: Fetcher, WKService {
                     }
                 }
             }
+        case (.PUT, .mediaWikiREST):
+            if let tokenType = mediaWikiRequest.tokenType {
+                requestMediaWikiAPIAuthToken(for: url, type: tokenType.wmfTokenType) { [weak self] result in
+                    switch result {
+                    case .success(let token):
+                        
+                        var tokenizedParams = request.parameters ?? [:]
+                        tokenizedParams["token"] = token.value
+                        
+                        self?.performPut(url: url, parameters: tokenizedParams, completion: completion)
```

Comment:
In the event that self is nil here, this call could leave the completion block uncalled. Would it make sense to call the completion block with some kind of failure here in the case that self is nil in order to prevent the caller from waiting on a completion that never occurs?

Reply:
@staykids Good catch - this is done!